client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java

```diff
@@ -804,6 +805,17 @@ static Request putTemplate(PutIndexTemplateRequest putIndexTemplateRequest) thro
         return request;
     }
 
+    static Request validateQuery(ValidateQueryRequest validateQueryRequest) throws IOException {
+        String endpoint = endpoint(validateQueryRequest.indices(), validateQueryRequest.types(),"_validate/query");
+        Request request = new Request(HttpGet.METHOD_NAME, endpoint);
+        Params params = new Params(request);
+        params.putParam("explain", Boolean.toString(validateQueryRequest.explain()));
+        params.putParam("all_shards", Boolean.toString(validateQueryRequest.allShards()));
+        params.putParam("rewrite", Boolean.toString(validateQueryRequest.rewrite()));
+        request.setEntity(createEntity(validateQueryRequest, REQUEST_BODY_CONTENT_TYPE));
```

Comment:
it looks like we might be missing a few more options such as `types` and `indicesOptions`, as specified in https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestValidateQueryAction.java#L62-L67

Reply:
`types` are being set in the URL I believe. I will check the `indicesOptions`.